group-coordinator/src/test/java/org/apache/kafka/coordinator/group/OffsetMetadataManagerTest.java

```diff
@@ -0,0 +1,1113 @@
+                ))
+            )
+        );
+    }
+
+    @Test
+    public void testGenericGroupOffsetCommitWhileInCompletingRebalanceState() {
+        OffsetMetadataManagerTestContext context = new OffsetMetadataManagerTestContext.Builder().build();
+
+        // Create an empty group.
+        GenericGroup group = context.groupMetadataManager.getOrMaybeCreateGenericGroup(
+            "foo",
+            true
+        );
+
+        // Add member.
+        group.add(new GenericGroupMember(
+            "member",
+            Optional.of("new-instance-id"),
+            "client-id",
+            "host",
+            5000,
+            5000,
+            "consumer",
+            new JoinGroupRequestData.JoinGroupRequestProtocolCollection(
+                Collections.singletonList(new JoinGroupRequestData.JoinGroupRequestProtocol()
+                    .setName("range")
+                    .setMetadata(new byte[0])
+                ).iterator()
+            )
+        ));
+
+        // Transition to next generation.
+        group.transitionTo(GenericGroupState.PREPARING_REBALANCE);
+        group.initNextGeneration();
+        assertEquals(1, group.generationId());
+
+        // Verify that the request is rejected with the correct exception.
+        assertThrows(RebalanceInProgressException.class, () -> context.commitOffset(
+            new OffsetCommitRequestData()
+                .setGroupId("foo")
+                .setMemberId("member")
+                .setGenerationIdOrMemberEpoch(1)
+                .setTopics(Collections.singletonList(
+                    new OffsetCommitRequestData.OffsetCommitRequestTopic()
+                        .setName("bar")
+                        .setPartitions(Collections.singletonList(
+                            new OffsetCommitRequestData.OffsetCommitRequestPartition()
+                                .setPartitionIndex(0)
+                                .setCommittedOffset(100L)
+                        ))
+                ))
+            )
+        );
+    }
+
+    @Test
+    public void testGenericGroupOffsetCommitWithoutMemberIdAndGeneration() {
+        OffsetMetadataManagerTestContext context = new OffsetMetadataManagerTestContext.Builder().build();
+
+        // Create an empty group.
+        GenericGroup group = context.groupMetadataManager.getOrMaybeCreateGenericGroup(
+            "foo",
+            true
+        );
+
+        // Add member.
+        group.add(new GenericGroupMember(
+            "member",
+            Optional.of("new-instance-id"),
+            "client-id",
+            "host",
+            5000,
+            5000,
+            "consumer",
+            new JoinGroupRequestData.JoinGroupRequestProtocolCollection(
+                Collections.singletonList(new JoinGroupRequestData.JoinGroupRequestProtocol()
+                    .setName("range")
+                    .setMetadata(new byte[0])
+                ).iterator()
+            )
+        ));
```

Comment:
nit: should we add a helper method? i think we only care about the member id and group instance id in these tests right